services/kinesis/src/it/java/software/amazon/awssdk/services/kinesis/SubscribeToShardIntegrationTest.java

```diff
@@ -290,8 +291,22 @@ private static void waitForStreamToBeActive() {
         Waiter.run(() -> asyncClient.describeStream(r -> r.streamName(streamName)).join())
               .until(b -> b.streamDescription().streamStatus().equals(StreamStatus.ACTIVE))
               .orFailAfter(Duration.ofMinutes(5));
-    }
 
+        // Additional verification to ensure stream is fully operational
+        Waiter.run(() -> {
+                  try {
+                      asyncClient.listShards(r -> r.streamName(streamName)).join();
+                      return true;
+                  } catch (Exception e) {
+                      if (e.getCause() instanceof ResourceInUseException) {
```

Comment:
We already have a Waiter waiting for the `ACTIVE` state, but we are failing because the stream is in the `CREATING` stage? That seems weird, are we sure this is helping?

Reply:
Yeap , I have attached the complete log in the internal issue.
The above log is taken from the codebuild logs.

Reply:
The additional check will ensure that the actual state is properly reflected in the system and will provide additional time with retry logic rather than immediately failing(same as we retrying the test case when it fails)
I recommend proceeding with this safe approach since the current failure of this test case is unnecessarily blocking the release and creating significant churn. This solution maintains reliability while addressing our immediate need to unblock the release process